### PR TITLE
Add compatibility with nodenv to the ios packager

### PR DESCRIPTION
**Pull request by migeorge**
nodenv is an alternative to nvm that I prefer to manage node verisons.

This PR adds nodenv initialization to the `react-native-xcode.sh` file in the packager, this allows the script to be run with the proper node shims in place.

This pull request should solve issue #4645

### CODE REVIEW

File: packager/react-native-xcode.sh [hunk 1]
```diff
@@ -41,6 +41,10 @@ elif [[ -x "$(command -v brew)" && -s "$(brew --prefix nvm)/nvm.sh" ]]; then
   . "$(brew --prefix nvm)/nvm.sh"
 fi
 
+# Initialize nodenv if it exists
+export PATH="$HOME/.nodenv/bin:$PATH"
```
Comment (ide) [inline packager/react-native-xcode.sh]: check if nodenv exists before modifying $PATH

Comment (migeorge) [inline packager/react-native-xcode.sh]: @ide, it should be fixed now

File: packager/react-native-xcode.sh [hunk 2]
```diff
@@ -41,6 +41,13 @@ elif [[ -x "$(command -v brew)" && -s "$(brew --prefix nvm)/nvm.sh" ]]; then
   . "$(brew --prefix nvm)/nvm.sh"
 fi
 
+# Initialize nodenv if it exists, if installed via git we must modify path
+if [[ -d "$HOME/.nodenv/bin" ]]; then
+  export PATH="$HOME/.nodenv/bin:$PATH"
+fi
+
+if which nodenv > /dev/null; then eval "$(nodenv init -)"; fi
```
Comment (ide) [inline packager/react-native-xcode.sh]: use `-x` and `command` instead of `which` (copy the style used for brew above)

**Comment by skevy** [comment]
DUPE of #4801...@ide if you'd be so kind to close whichever one you don't merge, that'd be great :)

**Comment by ide** [comment]
Closing since https://github.com/facebook/react-native/pull/4801 is merged.

**Pull request closed without merging**